src/rgw/driver/rados/rgw_rados.cc

```diff
@@ -5341,6 +5346,7 @@ int RGWRados::restore_obj_from_cloud(RGWLCCloudTierCtx& tier_ctx,
     return ret;
   }
 
+  attrs.erase(RGW_ATTR_TRANSITION_TIME);
```

Comment:
If this attr is being erased here, then it needs to be reset in "handle_obj_expiry".
That said, IMO, it may be better to retain this attr even for restored objects so that we can have timelog of transition, restore, expiry etc.